components/feature/toolbar/src/test/java/mozilla/components/feature/toolbar/WebExtensionToolbarTest.kt

```diff
@@ -4,53 +4,81 @@
 
 package mozilla.components.feature.toolbar
 
+import android.graphics.Bitmap
 import android.graphics.Color
-import android.graphics.drawable.Drawable
+import android.graphics.drawable.BitmapDrawable
 import android.view.View
 import android.widget.ImageView
 import android.widget.LinearLayout
 import android.widget.TextView
 import androidx.test.ext.junit.runners.AndroidJUnit4
+import kotlinx.coroutines.Dispatchers
+import kotlinx.coroutines.ExperimentalCoroutinesApi
+import kotlinx.coroutines.test.TestCoroutineDispatcher
+import kotlinx.coroutines.test.resetMain
+import kotlinx.coroutines.test.setMain
 import mozilla.components.concept.engine.webextension.BrowserAction
 import mozilla.components.support.base.android.Padding
+import mozilla.components.support.test.argumentCaptor
+import mozilla.components.support.test.ext.joinBlocking
 import mozilla.components.support.test.mock
 import mozilla.components.support.test.robolectric.testContext
 import mozilla.components.support.test.whenever
+import org.junit.After
 import org.junit.Assert.assertEquals
 import org.junit.Assert.assertFalse
 import org.junit.Assert.assertTrue
+import org.junit.Before
 import org.junit.Test
 import org.junit.runner.RunWith
 import org.mockito.Mockito.verify
 
 @RunWith(AndroidJUnit4::class)
 class WebExtensionToolbarTest {
+    private val testDispatcher = TestCoroutineDispatcher()
+
+    @Before
+    @ExperimentalCoroutinesApi
+    fun setUp() {
+        Dispatchers.setMain(testDispatcher)
+    }
+
+    @After
+    @ExperimentalCoroutinesApi
+    fun tearDown() {
+        Dispatchers.resetMain()
+        testDispatcher.cleanupTestCoroutines()
+    }
 
     @Test
     fun bind() {
-        val drawable: Drawable = mock()
+        val icon: Bitmap = mock()
         val imageView: ImageView = mock()
         val textView: TextView = mock()
         val view: View = mock()
 
         whenever(view.findViewById<ImageView>(R.id.action_image)).thenReturn(imageView)
         whenever(view.findViewById<TextView>(R.id.badge_text)).thenReturn(textView)
+        whenever(view.context).thenReturn(mock())
 
         val browserAction = BrowserAction(
             title = "title",
-            icon = drawable,
+            loadIcon = { icon },
             enabled = true,
             badgeText = "badgeText",
             badgeTextColor = Color.WHITE,
-            badgeBackgroundColor = Color.BLUE,
-            uri = "uri"
+            badgeBackgroundColor = Color.BLUE
         ) {}
 
         val action = WebExtensionToolbarAction(browserAction) {}
-
         action.bind(view)
+        action.iconJob?.joinBlocking()
+        testDispatcher.advanceUntilIdle()
+
+        val iconCaptor = argumentCaptor<BitmapDrawable>()
+        verify(imageView).setImageDrawable(iconCaptor.capture())
+        assertEquals(icon, iconCaptor.value.bitmap)
```

Comment:
nit: I'm not sure if it worths adding extra test coverage for the `iconJob?.cancel()`, what do you think?

Reply:
Yeah, might as well. Added a test now to make sure we cancel loading the icon.